src/main/kotlin/com/sourcegraph/cody/agent/protocol/ProtocolTextDocument.kt

```diff
@@ -71,6 +72,25 @@ private constructor(
       return ProtocolTextDocument(uri = uriFor(file), selection = getSelection(editor))
     }
 
+    @JvmStatic
+    fun fromEditorForDocumentEvent(editor: Editor, event: DocumentEvent): ProtocolTextDocument? {
+      val oldFragment = event.oldFragment.toString()
+      val file = FileDocumentManager.getInstance().getFile(editor.document) ?: return null
+      val startPosition = editor.offsetToLogicalPosition(event.offset).codyPosition()
+      val endCharacter =
```

Comment:
For `endCharacter` we cannot compute it using `editor.offsetToLogicalPosition` because document is already changed. Unfortunately IntelliJ gives us only offsets and not lines and columns, so we need to do a bit of calculations ourselves.